### PR TITLE
change tiberian to use the same method of save/load as redalert

### PR DESCRIPTION
**Pull request by capsterx**
The vtable hacking was not working on linux64 or the nintendo switch and I was not sure how to fix it.
So I switched the tiberian code to use the Read and then placement new with NoInitClass, which seems to be working.

### CODE REVIEW

File: common/noinit.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#pragma once
```
Comment (OmniBlade) [inline common/noinit.h]: We use ifdef guards in this project rather than pragma once.

**Review by OmniBlade — CHANGES_REQUESTED** [review]
Apart from the pragma once which to remain inkeeping with the rest of the code base should be an ifdef guard, the PR needs to be formatted with our clang format. I'll need to do some further testing to try and confirm it works on windows and will probably get @hifi to check linux, but overall this makes the code more similar between the two games and removes a nasty hack so I'd like to work towards getting this merged.

File: common/noinit.h (hunk 1 above)
Comment (capsterx) [reply common/noinit.h]: I updated it to use header guards

**Comment by capsterx** [comment]
This somehow breaks buildings displaying and sidebar and I cannot figure out where that happens, it even breaks for brand new games after a load.

**Comment by capsterx** [comment]
> This somehow breaks buildings displaying and sidebar and I cannot figure out where that happens, it even breaks for brand new games after a load.

I have found part of the sidepart part, but not the building part.

**Comment by tomsons26** [comment]
This PR is scary af, ngl, but it is necessary to do as the save load system is the one of the main blockers to commonizing quite a bit of game code.

**Comment by capsterx** [comment]
> > This somehow breaks buildings displaying and sidebar and I cannot figure out where that happens, it even breaks for brand new games after a load.
> 
> I have found part of the sidepart part, but not the building part.

I found the building part, it was missing a NoInit of a superclass on buildings.   It seems to be working correctly now.

File: redalert/function.h [hunk 2]
```diff
@@ -149,8 +131,8 @@ Map(screen) class heirarchy.
 #include "shastraw.h"
 #include "rndstraw.h"
 
-// Should be part of WWLIB.H. This is used in JSHELL.CPP.
-typedef struct
+          // Should be part of WWLIB.H. This is used in JSHELL.CPP.
```
Comment (OmniBlade) [inline redalert/function.h]: Odd whitespacing here? Formatting issue?

**Review by OmniBlade — CHANGES_REQUESTED** [review]
This needs rebasing on the latest vanilla branch as noinit was moved to common in another PR already I believe. I've done some testing on msvc 64bit builds and it appears to be working fine, unless @tomsons26 or @hifi have any objections I'll be happy to merge onces the rebase is done.

File: redalert/function.h (hunk 2 above)
Comment (capsterx) [reply redalert/function.h]: Not sure, I ran clang-format over it, but I think I fixed it in the latest push

**Comment by hifi** [comment]
I'm fine with this after conflicts are resolved and rebased. :+1:

**Comment by capsterx** [comment]
Rebased

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**